Emby.Server.Implementations/Library/DefaultAuthenticationProvider.cs

```diff
@@ -19,31 +20,110 @@ public DefaultAuthenticationProvider(ICryptoProvider crypto)
 
         public bool IsEnabled => true;
 
+
+        //This is dumb and an artifact of the backwards way auth providers were designed.
+        //This version of authenticate was never meant to be called, but needs to be here for interface compat
+        //Only the providers that don't provide local user support use this
         public Task<ProviderAuthenticationResult> Authenticate(string username, string password)
         {
             throw new NotImplementedException();
         }
 
-        public Task<ProviderAuthenticationResult> Authenticate(string username, string password, User resolvedUser)
-        {
-            if (resolvedUser == null)
-            {
-                throw new Exception("Invalid username or password");
-            }
-
-            var success = string.Equals(GetPasswordHash(resolvedUser), GetHashedString(resolvedUser, password), StringComparison.OrdinalIgnoreCase);
 
-            if (!success)
-            {
-                throw new Exception("Invalid username or password");
-            }
+        //This is the verson that we need to use for local users. Because reasons.
+        public Task<ProviderAuthenticationResult> Authenticate(string username, string password, User resolvedUser)
+        {
+            ConvertPasswordFormat(resolvedUser);
+            byte[] passwordbytes = Encoding.UTF8.GetBytes(password);
+            bool success = false;
+            if (resolvedUser == null)
+            {
+                success = false;
+                throw new Exception("Invalid username or password");
+            }
+            if (!resolvedUser.Password.Contains("$"))
+            {
+                ConvertPasswordFormat(resolvedUser);
+            }
+            PasswordHash ReadyHash = new PasswordHash(resolvedUser.Password);

```

Comment:
pls be consistent with case in naming, either `ThisCase` or `thatCase`, but not a mix of two... I suggest to keep all non-classtype things (like instances) in `thatCase`.

Reply:
I have tried to correct this in general. Reopen if I've missed some, because I probably have.